aip/general/0129.md

```diff
@@ -0,0 +1,150 @@
+---
+id: 129
+state: draft
+created: 2023-06-21
+placement:
+  category: resource-design
+  order: 65
+---
+
+# Server-Modified Values and Defaults
+
+Server-modified and default values often result in a poor user experience for
+[state-driven clients](state-driven-clients). These clients are often unable to
+tell when their desired state matches the current state for these fields, as the
+rules by which a server may modify and return values are complex, not public,
+and not repeatable. This AIP aims to clarify what behaviors can be expected from
+services and describing how they can be repeated on the client-side.
+
+## Guidance
+
+### Single Owner Fields
+
+Fields should have a single owner, whether that is the client or the server.
+Server owned fields should be indicated with the `OUTPUT_ONLY` field_behavior.
+All other types of fields should be considered to be owned by the client. The
+server should respect the value (or lack thereof) for all client owned fields
+and not modify them. Failure to respect client set values can lead to
+[Declarative clients][] entering an infinite loop to correct the change.
+
+### Effective Values
+
+There are instances where a service will allocate, generate, or calculate a
+value if the client chooses not to specify one. For example: a client may
+create a virtual machine without specifying a static IP address for the virtual
+machine to be available on. Such a scenario is opting into dynamic IP address
```

Comment:
There seems an important and subtle distinction between this scenario (in which the system ends up managing the address), and a related scenario in which the system assigns an IP address on behalf of the client, but in which the IP address remains fixed (i.e., has the same semantic as if the caller had selected that address).

Reply:
Those are different situations, but in my mind are handled in the same way. By splitting the client input and the server value for a field that can be dynamic or statically assigned by the server if not specified by the client I think both problems are solved